dataland-frontend/tests/e2e/specs/sfdr/DataIntegrity.ts

```diff
@@ -0,0 +1,101 @@
+before(function () {
+  cy.fixture("CompanyInformationWithSfdrPreparedFixtures").then(function (jsonContent) {
+    const sfdrPreparedFixtures = jsonContent as Array<FixtureData<SfdrData>>;
+    testSfdrCompany = getPreparedFixture("CompanyInformationWithSfdrData", sfdrPreparedFixtures);
+  });
+});
+describeIf(
+  "As a user, I expect that the upload form works correctly when editing and uploading a new SFDR dataset",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+  },
+  function () {
+    beforeEach(() => {
+      cy.ensureLoggedIn(admin_name, admin_pw);
+    });
+
+    //TODO Rename file to DataIntegrity after everything is finished
+    /**
+     * validates that the data uploaded via the function `uploadSfdrDataViaForm` is displayed correctly for a company
+     * @param companyId the company associated to the data uploaded via form
+     */
+    function validateFormUploadedData(companyId: string): void {
+      cy.visit("/companies/" + companyId + "/frameworks/" + DataTypeEnum.Sfdr);
+      cy.get(".p-datatable-tbody").find(".p-rowgroup-header").eq(0).should("have.text", "General");
+
+      cy.get(".p-datatable-tbody").find(".p-rowgroup-header").eq(2).should("have.text", "Biodiversity");
+      toggleRowGroup("biodiversity");
+      cy.contains("td.headers-bg", "Primary Forest And Wooded Land Of Native Species Exposure").should("exist");
+      cy.contains("td.headers-bg", "Primary Forest And Wooded Land Of Native Species Exposure")
+        .siblings("td")
+        .should("have.text", "Yes");
+      cy.contains("td.headers-bg", "Protected Areas Exposure").should("exist");
+      cy.contains("td.headers-bg", "Protected Areas Exposure").siblings("td").should("have.text", "No");
+      cy.contains("td.headers-bg", "Rare Or Endangered Ecosystems Exposure").should("exist");
+      cy.contains("td.headers-bg", "Rare Or Endangered Ecosystems Exposure").siblings("td").should("have.text", "Yes");
+    }
+
+    /**
+     * Set the quality for the sfdr test dataset
+     */
+    function setQualityInSfdrUploadForm(): void {
+      cy.get('[data-test="primaryForestAndWoodedLandOfNativeSpeciesExposure"]')
+        .find('select[name="quality"]')
+        .select(3);
+      cy.get('[data-test="protectedAreasExposure"]').find('select[name="quality"]').select(3);
+      cy.get('[data-test="rareOrEndangeredEcosystemsExposure"]').find('select[name="quality"]').select(3);
+    }
+    it("Create a company via api and upload a SFDR dataset via the api", () => {
+      const uniqueCompanyMarker = Date.now().toString();
+      const testCompanyName = "Company-Created-In-Sfdr-DataIntegrity-Test-" + uniqueCompanyMarker;
+      getKeycloakToken(admin_name, admin_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompanyName)).then((storedCompany) => {
+          return uploadOneSfdrDataset(token, storedCompany.companyId, "2021", testSfdrCompany.t).then(
```

Comment:
There is a method "uploadCompanyAndSfdrDataViaApi" which combines these two into one method. It is in the file `SfdrUpload.ts` and is currently grey (unused) according to the IDE.
So we should either use it, or remove it.

Reply:
Taking care of it

Reply:
uploadCompanyAndSfdrDataViaApi

Reply:
👍